transportable-udfs-avro/src/main/java/com/linkedin/transport/avro/AvroWrapper.java

```diff
@@ -43,27 +43,69 @@ private AvroWrapper() {
 
   public static StdData createStdData(Object avroData, Schema avroSchema) {
     switch (avroSchema.getType()) {
-      case INT:
+      case INT: {
+        if (!(avroData instanceof Integer)) {
+          throw new IllegalArgumentException("Unsupported type for Avro integer: " + avroData.getClass());
+        }
         return new AvroInteger((Integer) avroData);
-      case LONG:
+      }
+      case LONG: {
+        if (!(avroData instanceof Long)) {
+          throw new IllegalArgumentException("Unsupported type for Avro long: " + avroData.getClass());
+        }
         return new AvroLong((Long) avroData);
-      case BOOLEAN:
+      }
+      case BOOLEAN: {
+        if (!(avroData instanceof Boolean)) {
+          throw new IllegalArgumentException("Unsupported type for Avro boolean: " + avroData.getClass());
+        }
         return new AvroBoolean((Boolean) avroData);
-      case STRING:
-        return new AvroString((Utf8) avroData);
-      case FLOAT:
+      }
+      case STRING: {
+        if (avroData instanceof Utf8) {
+          return new AvroString((Utf8) avroData);
+        } else if (avroData instanceof String) {
+          return new AvroString(new Utf8((String) avroData));
```

Comment:
Note that this [PR](https://github.com/linkedin/transport/pull/34/files#diff-6275c5464d6fa3ca341b65556e1e92fe80e487b42412f01ff038e9a835fc36c5R39) handles this case better. I will make some time soon to merge it :)